src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -212,6 +220,19 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context, Abs
                 return new BitsOrPostingList(new ArrayPostingList(postings.toIntArray()));
             }
 
+            if (graph instanceof CassandraDiskAnn && ((CassandraDiskAnn) graph).getCompressedVectors() != null)
```

Comment:
This needs to go in the "rows <= maxBruteForceRows" block, or it will never perform the actual index search

the logic here is tangled because we've smashed the two possible paths into a single return class.  return BitsOrPostingList(PostingList) -> these are the row ids, no search is performed.  return BOPL(Bits) -> too many to brute force, we put the valid graph ordinals in a Bits and caller needs to do a search with it